extension/content/components/recipes/form/arguments/PreferenceRollout.js

```diff
@@ -0,0 +1,211 @@
+        >
+          Add Preference
+        </IconButton>
+      </Panel>
+    </FormGroup>
+  );
+}
+
+function Preference({ index }) {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+  const preference = data.arguments.preferences[index];
+  const { preferences } = data.arguments;
+  const [preferenceType, setPreferenceType] = React.useState(
+    typeof preference.value,
+  );
+  const handleClickDelete = () => {
+    dispatch({
+      type: ACTION_UPDATE_DATA,
+      data: {
+        ...data,
+        arguments: {
+          ...data.arguments,
+          preferences: preferences.filter((_, i) => i !== index),
+        },
+      },
+    });
+  };
+
+  const handleChange = (name, cb = (v) => v) => {
+    return (value) => {
+      dispatch({
+        type: ACTION_UPDATE_DATA,
+        data: {
+          ...data,
+          arguments: {
+            ...data.arguments,
+            preferences: preferences.map((p, i) => {
+              if (i === index) {
+                return { ...p, [name]: cb(value) };
+              }
+
+              return p;
+            }),
+          },
+        },
+      });
+    };
+  };
+
+  const parseNumericInput = (value) => {
+    if (!value) {
+      return "";
+    }
+
+    return parseInt(value, 10);
+  };
+
+  let valueField = (
+    <Input block value={preference.value} onChange={handleChange("value")} />
+  );
+
+  if (preferenceType === "number") {
+    valueField = (
+      <InputNumber
+        block
+        value={preference.value}
+        onChange={handleChange("value", parseNumericInput)}
+      />
+    );
+  } else if (preferenceType === "boolean") {
+    valueField = (
+      <RadioGroup
+        inline
+        value={preference.value}
+        onChange={handleChange("value")}
+      >
+        <Radio value={true}>True</Radio>
+        <Radio value={false}>False</Radio>
+      </RadioGroup>
+    );
+  }
+
+  return (
+    <FormGroup>
+      <div className="d-flex">
+        <div className="pr-2">
+          <FormGroup>
+            <ControlLabel>&nbsp;</ControlLabel>
+            <IconButton
+              circle
+              color="red"
+              icon={<Icon icon="trash" />}
+              size="sm"
+              onClick={handleClickDelete}
+            />
+          </FormGroup>
+        </div>
+
+        <div className="pr-1">
+          <FormGroup>
+            <ControlLabel>Name</ControlLabel>
+            <Input
+              style={{
+                width: "400px",
+              }}
+              value={preference.preferenceName}
+              onChange={handleChange("preferenceName")}
+            />
+          </FormGroup>
+        </div>
+        <div className="pr-1">
+          <FormGroup>
+            <ControlLabel>Preference Type</ControlLabel>
+            <InputPicker
+              data={PREFERENCE_TYPE_OPTIONS}
+              value={preferenceType}
+              onChange={(value, _) => {
```

Comment:
i've noticed this in a couple PRs... is there a reason for the underscore? if you're not using it you don't need to include it here. you could just do `(value) =>`

Reply:
also i would move this up to something like:
```js
const handlePreferenceTypeChange = (value) => {
```